lib/dice/src/mfg.rs

```diff
@@ -229,13 +248,15 @@ impl DiceMfg for SerialMfg<'_> {
                 }
                 MfgMessage::CsrPlz => self.handle_csrplz(),
                 MfgMessage::IdentityCert(cert) => {
-                    self.handle_persistid_cert(cert)
+                    self.handle_persistid_cert(cert, &hash)
```

Comment:
Not thrilled with having to carry hash through various method calls solely so it can be used as part of `send_ack()`.  Adopting TLS 1.3's transcript hash approach (https://datatracker.ietf.org/doc/html/rfc8446#section-4.4.1) seems like it would simplify the code here and reduce the traffic sent over the wire.  Essentially, `SerialMfg` would keep a SHA3 hasher and each received and sent message would be added to the hash.  When an Ack is sent (or any other message that wishes to include a hash to check session integrity at that point), `finalize_reset()` gets called on the hasher.  `send_ack` now only needs a `&mut self` and integrity can be checked over a run of multiple request/response pairs.

Reply:
WE actually do this in sprockets: https://github.com/oxidecomputer/sprockets/blob/7c9bcb262cb117795976056a9102b96c9971d80b/session/src/client.rs#L360

Reply:
Thanks for the review @kc8apf. Passing the hash through parameters wasn't so much an artifact of the hashing approach taken as the implementation. I've gone back over the code to address these concerns and the flash savings + simplification are significant.